js-advanced-functions/fn-exercise-1.js

```diff
@@ -0,0 +1,53 @@
+//1. part one
+let initialData=[200,500,300,1000]
+console.log('INITIAL DATA: ',initialData)
+console.log('Part 1 of the exercise -------------')
+
+const incrementing=(array,padding)=>{
+    const newArray=[];
+    for(i=0;i<array.length;i++){
+        newArray.push(array[i]+padding)
+    }
+    // for(el of array){
+    //     el=el+padding //remove this
+    //     newArray.push(el+padding) //same to this
+
+    // }
+    // the two for cycles do the same
+    console.log('Incrementing function by 2',newArray)
+}
+incrementing(initialData,2)
+
+
+const decrementing=(array,padding)=>{
+    const newArray2=[]
+    for(i=0;i<array.length;i++){
+        newArray2.push(array[i]-padding)
+    }
+    console.log('Decrementing function by 2',newArray2)
+}
+decrementing(initialData,2)
+
+//2.part two 
+//HOF
+console.log('Part 2 of the exercise --------------')
+
+const operationInc = (number,padding)=>{
+    return number+padding       
+}
+// 
+const operationDec = (number,padding)=>{
+    return number-padding    
+}
+const main =(array,fn,padding)=>{
+    const newArray=[]
+    for(const elem of array){
+        newArray.push(fn(elem,padding))
+    }
+    return newArray
+}
+const incrHOF=main(initialData,operationInc,3)
+const decrHOF=main(initialData,operationDec,3)
+console.log(incrHOF)
+console.log(decrHOF)
+// Where to print the message `HOF call (increment) by 3` and `HOF call (increment) by 3` ?
```

Comment:
You did everything well, just the final part.
`const main = (array, fn, padding) => {
    const newArray = []
    for (const elem of array) {
        newArray.push(fn(elem, padding))
    }
    return (fn) => { // here you return an array but you need to return new function
        return fn(newArray); // this function will be our sort function
    }
}
const incrHOF = main(initialData, operationInc, 3)
const decrHOF = main(initialData, operationDec, 3)
console.log(incrHOF(sort));
console.log(decrHOF(sort))`
What is left here is to create a sort function and use it in the console logs above. This is the way how HOF are working